generator/src/test/java/com/scottlogic/deg/generator/cucumber/OfLength.feature

```diff
@@ -1,593 +1,455 @@
-
-
-  Scenario Outline: Running an 'ofLength' request that includes a negation of a valid numeric length should be successful
-    Given foo is anything but of length <length>
-    And foo is in set:
-      | "a"  |
-      | "aa" |
-    Then the following data should be generated:
-      | foo        |
-      | null       |
-      | <expected> |
-    And the following data should not be included in what is generated:
-      | foo           |
-      | <notExpected> |
-  Examples:
-      | length | expected    | notExpected |
-      | 1      | "aa"        | "a"         |
-      | 2      | "a"         | "aa"        |
-
-
-  Scenario Outline: Running an 'ofLength' request on a roman alphabet character string value and white space should be successful
-    Given foo is of length <length>
-    And foo is in set:
-      | "a"          |
-      | "  10  10  " |
-    Then the following data should be generated:
-      | foo        |
-      | null       |
-      | <expected> |
-    Examples:
-      | length | expected     |
-      | 10     | "  10  10  " |
-
-
-  Scenario: Running an 'ofLength' request that includes a numeric value = 0 on an empty string should be successful
-    Given foo is of length 0
-    And foo is in set:
-      | ""  |
-      | "a" |
-    Then the following data should be generated:
-      | foo  |
-      | null |
-      | ""   |
+Background:
+     Given the generation strategy is full
+       And there is a field foo
+       And foo is of type "string"
+
+Scenario Outline: Running an 'ofLength' request on a roman alphabet character string should be successful
+     Given foo is of length <length>
+       And foo is in set:
+         | "a"  |
+         | "aa" |
+     Then the following data should be generated:
+       | foo        |
+       | null       |
+       | <expected> |
+       And the following data should not be included in what is generated:
+         | foo           |
+         | <notExpected> |
+     Examples:
+       | length | expected | notExpected |
+       | 1      | "a"      | "aa"        |
+       | 2      | "aa"     | "a"         |
+       | 1.0    | "a"      | "aa"        |
+
+Scenario Outline: Running an 'ofLength' request that includes a negation of a valid numeric length should be successful
+     Given foo is anything but of length <length>
+       And foo is in set:
+         | "a"  |
+         | "aa" |
+     Then the following data should be generated:
+       | foo        |
+       | null       |
+       | <expected> |
+       And the following data should not be included in what is generated:
+         | foo           |
+         | <notExpected> |
+     Examples:
+       | length | expected    | notExpected |
+       | 1      | "aa"        | "a"         |
+       | 2      | "a"         | "aa"        |
+
+Scenario Outline: Running an 'ofLength' request on a roman alphabet character string value and white space should be successful
```

Comment:
Do we need an "Outline" if we have just one "Example"

Reply:
This test could be reformatted but wasn't for the sake of expediency. Do you think it would benefit understandability from being reformatted @MattCline-SL ?

Reply:
Test reformatted in line with comment

Reply:
Ah the PR made it look like a new test written, I agree not necessary to do here :)